packages/search/src/content/index.ts

```diff
@@ -89,13 +89,195 @@ export class ContentSearchService
+ * // Search for all public and private content associated with a site
+ * /////////////////////////////////////////////////////////////////////
+ * const options: IContentSearchOptions = {
+ *    site: 'https://my-site.hub.arcgis.com',
+ *    portal: 'https://www.arcgis.com',
+ *    // Note: only private content visible to the authenticated user will be returned
+ *    authentication: new UserSession(...),
+ * }
+ * const searchResults = await searchContent({ options });
+ * ...
+ *
+ * ///////////////////////////////////////////////////////////
+ * // Search for all public content associated with a site
+ * ///////////////////////////////////////////////////////////
+ * const options: IContentSearchOptions = {
+ *    site: 'https://my-site.hub.arcgis.com',
+ *    portal: 'https://www.arcgis.com',
+ *    authentication: new UserSession(...),
+ * }
+ * const filter: IContentSearchFilter = {
+ *    access: 'public' // Can also be expressed as { value: ['public'] }
+ * }
+ * const searchResults = await searchContent({ filter, options });
+ * ...
  *
- * const searchResults = searchContent({ filters, options })
+ * //////////////////////////////////////////////////////////
+ * // Search for all private content associated with a site
+ * ///////////////////////////////////////////////////////////
+ *
+ * // TODO: As of 8/17/21 the hub api has a bug in which
+ * // Any query with a not filter on the access field
+ * // (e.g access: not('public')) will return no results
+ *
+ * const options: IContentSearchOptions = {
+ *    site: 'https://my-site.hub.arcgis.com',
+ *    portal: 'https://www.arcgis.com',
+ *    // Note: only private content visible to the authenticated user will be returned
+ *    authentication: new UserSession(...),
+ * }
+ * const filter: IContentSearchFilter = {
+ *    access: {
+ *        value: ['public'],
+ *        bool: IBooleanOperator.NOT
+ *    }
+ * }
+ * const searchResults = await searchContent({ filter, options });
+ *
+ * ...
+ * /////////////////////////////////
+ * // Search for a specific item
+ * /////////////////////////////////
+ * const options: IContentSearchOptions = {
+ *    portal: 'https://www.arcgis.com',
+ *    authentication: new UserSession(...),
+ * }
+ * const filter: IContentSearchFilter = {
+ *    id: 'my_item_id' // Can also be expressed as { value: ['my_item_id'] }
+ * }
+ * const searchResults = await searchContent({ filter, options });
  * ```
+ * There are a couple gotchas that need to be accounted for:
+ *
+ * 1) There is no way to specify the number of results per page. If you need to fetch all
+ * items that match a given query, you'll need to utilize the `hasNext` flag as well as the
+ * `next` function included on the return object and make multiple XHR requests.
+ *
+ * 2) There are differences in item structure for results returned from the
+ * hub index vs AGO/Enterprise (i.e. 'public' items vs 'private' items).
+ * Here is an example of a 'private' map result:
```

Comment:
This section was added as per @ajturner's comment on the issue:

> How do I search a Site?
Does this include both public + private content?
What are the differences in metadata attributes?

I couldn't find an easy way to see exactly what was different between the two and I'm not sure which attribute differences are significant (other than obvious ones like `slug` and `composeStatus`). We have to remember as well that https://devtopia.esri.com/dc/hub/issues/1713 will change the results into `IHubContents` in the near future

Reply:
Unfortunately, this seems to be par for the course. There are a default list of attributes that are returned, but these are incomplete compared to Portal API. See [schema.json](https://github.com/ArcGIS/hub-indexer/blob/master/packages/api/schema.json) for how the API properties of the V3 API are mapped to the elasticsearch document properties. Those that appear to be identical attributes to their Portal API equivalents, but do not come from the `item` ES document property, may be different.